code/game/objects/items/weapons/tools/pickaxe.dm

```diff
@@ -9,90 +9,136 @@
 	item_state = "pickaxe"
 	w_class = ITEM_SIZE_LARGE
 	matter = list(MATERIAL_STEEL = 6)
-	tool_qualities = list(QUALITY_DIGGING = 30, QUALITY_PRYING = 20, QUALITY_EXCAVATION = 10)
+	tool_qualities = list(QUALITY_EXCAVATION = 10, QUALITY_PRYING = 20) //So it still shares its switch off quality despite not yet being used.
+	switched_off_qualities = list(QUALITY_EXCAVATION = 10, QUALITY_PRYING = 20)
+	switched_on_qualities = list(QUALITY_DIGGING = 30, QUALITY_PRYING = 20)
+	toggleable = TRUE
 	origin_tech = list(TECH_MATERIAL = 1, TECH_ENGINEERING = 1)
 	attack_verb = list("hit", "pierced", "sliced", "attacked")
-	sharp = 1
-	structure_damage_factor = 3 //Drills and picks are made for getting through hard materials
+	sharp = TRUE
+	structure_damage_factor = STRUCTURE_DAMAGE_BORING //Drills and picks are made for getting through hard materials
 	//They are the best anti-structure melee weapons
 	embed_mult = 1.2 //Digs deep
+	mode = EXCAVATE //Mode should be whatever is the starting tool and off quality.
+
+/obj/item/weapon/tool/pickaxe/pickup(mob/user)
```

Comment:
```suggestion
/obj/item/weapon/tool/pickaxe/equip(mob/user)
```

Reply:
`equip` is not a defined proc, did you mean something else?

Reply:
Oh yes, sorry. It should be
`/obj/item/weapon/tool/pickaxe/equipped(mob/user)`
`pickup(...)` calls only when user pick smth with clicking on it. It's usefull for picking animation for example.
`equipped(...)` calls when user equip items without any conditions. For example - passing items with **give** verb or undress menu.

```suggestion
/obj/item/weapon/tool/pickaxe/equipped(mob/user)
```